### PR TITLE
implement pin lock option

### PR DESCRIPTION
**Pull request by guerrap**
Done:

- option to configure the pin in the `Settings` view

### CODE REVIEW

File: src/views/app/index.js [hunk 1]
```diff
@@ -50,54 +54,95 @@ const themes = {
...
+        AppState.addEventListener('change', handleAppStateChange);
+
+        return () => {
+            AppState.removeEventListener('change', handleAppStateChange);
+        };
+    }, []);
+
     useEffect(() => {
         SplashScreen.hide();
+        dispatch(triggerPinVerify());
```
Comment (guerrap) [inline src/views/app/index.js:84]: Probably not needed, since it's already called in `handleAppStateChange`. I'll check later

**Comment by guerrap** [comment]
Done:

- fixed an issue where the user could see the portfolio data for a second, before being redirected to the pin lock view
- hide the app view while in the background, and prevent the user from taking screenshots (Android only)

**Comment by guerrap** [comment]
Done:

- if the pin is incorrect clear the pin displayed so that the user doesn't have to manually delete the values

**Comment by luzzif** [comment]
Awesome, thanks again.

**Pull request merged**